src/plone/app/testing/layers.py

```diff
@@ -382,19 +384,21 @@ class MockMailHostLayer(Layer):
     """
     defaultBases = (PLONE_FIXTURE,)
 
-    def setUp(self):
+    def testSetUp(self):
         with zope.zopeApp() as app:
             portal = app[PLONE_SITE_ID]
-            portal.email_from_address = 'noreply@example.com'
-            portal.email_from_name = 'Plone Site'
+            registry = getUtility(IRegistry, context=portal)
+            if not registry["plone.email_from_address"]:
+                registry["plone.email_from_address"] = "noreply@example.com"
+            if not registry["plone.email_from_name"]:
+                registry["plone.email_from_name"] = u"Plone site"
```

Comment:
I have a small reservation: this changes the registry, but does not undo this after test teardown.
But that was already the case for the properties on the portal, so I guess it is no big deal.

Reply:
I actually want to do that!
Thanks anyway!